lib/elixir/src/elixir_clauses.erl

```diff
@@ -14,7 +14,7 @@ match(Fun, Expr, #{context := Context, match_vars := Match, prematch_vars := nil
   {EExpr, EE#{context := Context, match_vars := Match, prematch_vars := nil}}.
 
 def({Meta, Args, Guards, Body}, #{match_vars := Match} = E) ->
-  {EArgs, EA}   = elixir_expand:expand(Args, E#{context := match, match_vars := []}),
+  {EArgs, EA}   = elixir_expand:expand(Args, E#{context := match, match_vars := warn}),
```

Comment:
My intuition says that match_vars should be `[]`, because they are always an empty list inside of a match. Maybe we need another solution for the problem?